tests/test_kto_trainer.py

```diff
@@ -171,20 +171,18 @@ def test_tokenize_and_process_tokens(self):
                 self.assertListEqual(tokenized_dataset["answer_input_ids"][0], [5968, 1219, 72, 3621, 284, 1826, 345])
                 self.assertListEqual(tokenized_dataset["answer_attention_mask"][0], [1, 1, 1, 1, 1, 1, 1])
 
-                # Test reversal of (prompt, completion) pairs for KL dataset
-                tokenized_kl_dataset = tokenized_dataset.map(_get_kl_dataset, batched=True, batch_size=2)
-                self.assertListEqual(
```

Comment:
What are you talking about? 
- The fact that `"prompt_input_ids"` is not altered? 
    I didn't understand the point of this test, given that we don't touch the prompts, but I can add a test, it's not a big deal.
- The fact that index 1 of the new dataset is equal to index 0 of the old one. 
    This test seems to me to be exactly what you shouldn't do, because that's not what you really want to test: you want this function to create mismatches.

Reply:
first point done in [9fff4a8](https://github.com/huggingface/trl/pull/1974/commits/9fff4a8cab34b0c4628c09b954bcb6dad736167f)

Reply:
Ah I now understand what you're doing - the PR adds a new test and removes some old tests which were indeed bad ones. (Next time just mention it in the PR description or add a comment on the PR)

Feel free to revert https://github.com/huggingface/trl/pull/1974/commits/9fff4a8cab34b0c4628c09b954bcb6dad736167f  since you're correct the old tests weren't testing the intended behaviour.